### PR TITLE
rename cds into mocserver

### PR DESCRIPTION
**Pull request by ManonMarchand**
This PR is linked to the discussion we had in #2764 
Please comment because I was unsure about the best way to inform of the deprecation without breaking API too much. 

For now, the behaviour is that when a user does : 

```python
>>> from astroquery.cds import cds
```

There is a warning but the class is still imported with its old name: 

```python
<stdin>:1: DeprecationWarning: The ``cds`` module has been moved to astroquery.mocserver,and ``CdsClass`` has been renamed ``MOCserver``. It means that 'from astroquery.cds import cds' became 'from astroquery.mocserver import mocserver'. Please update your imports.
>>> cds.
cds.TIMEOUT                 cds.URL                     cds.cache_location          cds.clear_cache()           cds.find_datasets(          cds.name                    cds.path_moc_file           cds.query_region(           cds.query_region_async(     cds.reset_cache_location()  cds.return_moc
```

However, if the import is done with `import astroquery` only, the users code will be broken. 
Is there a better way to do this? 

Thanks a lot for everything astroquery does <3

PS: the MOCserver module will receive some love from CDS soon because there is now the possibility to query TimeMOC and FrequencyMOC on our side and we'd like to update the library here too

### CODE REVIEW

**Comment by bsipocz** [comment]
> PS: the MOCserver module will receive some love from CDS soon because there is now the possibility to query TimeMOC and FrequencyMOC on our side and we'd like to update the library here too

❤️

**Review by bsipocz — COMMENTED** [review]
I have some minor comments mostly about naming things, otherwise it all looks good. 

If you have some time for a follow-up, definitely not part of this PR: I've noticed that the docs is still skipped for testing, fixing that would be superb. (other modules can be used as an example, basically removing the skip directives and using the doctest-remote-data directive, and in the meantime fixing any non-matching outputs).

Also, I noticed one of the remote tests run for a rather long time, I wonder whether it could cut back to something quicker/smaller? (to run all the tests for this module took me 8.5 minutes)

File: astroquery/mocserver/core.py [hunk 1]
```diff
@@ -19,20 +19,18 @@
...
 
 try:
     from regions import CircleSkyRegion, PolygonSkyRegion
 except ImportError:
     print("Could not import astropy-regions, which is a requirement for the CDS service."
           "Please refer to http://astropy-regions.readthedocs.io/en/latest/installation.html for how to install it.")
-    pass
 
-__all__ = ['cds', 'CdsClass']
+__all__ = ['mocserver', 'MOCserver']
```
[the remark this answers is not in the document]
Comment (ManonMarchand) [reply astroquery/mocserver/core.py:29]: Sure! I was unaware of the naming conventions. Thanks for the review I'll address the comments soon

**Comment by ManonMarchand** [comment]
> I have some minor comments mostly about naming things, otherwise it all looks good.

Thanks for the review! I think I addressed your comments. Let me know if I got them wrong.

> If you have some time for a follow-up, definitely not part of this PR: I've noticed that the docs is still skipped for testing, fixing that would be superb. (other modules can be used as an example, basically removing the skip directives and using the doctest-remote-data directive, and in the meantime fixing any non-matching outputs).
> 
> Also, I noticed one of the remote tests run for a rather long time, I wonder whether it could cut back to something quicker/smaller? (to run all the tests for this module took me 8.5 minutes)

Definitely adding these two things to my list. Won't be before the end of the summer holiday though.

File: astroquery/cds/__init__.py [hunk 2]
```diff
@@ -18,27 +18,14 @@
...
-    timeout = _config.ConfigItem(
-        30,
-        'Time limit for connecting to template_module server.')
-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import warnings
 
-conf = Conf()
+warnings.warn("The ``cds`` module has been moved to astroquery.mocserver,"
+              "and ``CdsClass`` has been renamed ``MOCServerClass``. "
```
Comment (bsipocz) [inline astroquery/cds/__init__.py:25]:
```suggestion
warnings.warn("The ``cds`` module has been moved to astroquery.mocserver, "
              "and ``CdsClass`` has been renamed to ``MOCServerClass``. "
```

File: astroquery/cds/__init__.py [hunk 3]
```diff
@@ -18,27 +18,14 @@
...
+              "Please update your imports.", DeprecationWarning, stacklevel=2)
 
-from .core import cds, CdsClass
+from astroquery.mocserver import MOCServer as cds
+from astroquery.mocserver import MOCServerClass as CdsClass
 
-__all__ = ['cds', 'CdsClass',
-           'Conf', 'conf',
-           ]
+__all__ = ['conf', 'Conf', 'cds', 'CdsClass']
```
Comment (bsipocz) [inline astroquery/cds/__init__.py:31]: conf anf Conf need to be imported, too

File: astroquery/cds/__init__.py [hunk 4]
```diff
@@ -18,27 +18,14 @@
...
+import warnings
 
-conf = Conf()
+warnings.warn("The ``cds`` module has been moved to astroquery.mocserver,"
+              "and ``CdsClass`` has been renamed ``MOCServerClass``. "
+              "Please update your imports.", DeprecationWarning, stacklevel=2)
 
-from .core import cds, CdsClass
+from astroquery.mocserver import MOCServer as cds
+from astroquery.mocserver import MOCServerClass as CdsClass
```
Comment (bsipocz) [inline astroquery/cds/__init__.py:29]:
```suggestion
from astroquery.mocserver import MOCServerClass as CdsClass
from astroquery.mocserver import conf, Conf
```

**Review by bsipocz — APPROVED** [review]
A few very minor suggestions that I'll just directly commit to the branch, and then merge when we get the new green CI status.

Thank you very much for the quick turnaround with the PR!

**Comment by bsipocz** [comment]
`pip install -U --pre astroquery` should get you the new module :)

**Pull request merged**